tenant/src/main/scala/com/improving/app/tenant/domain/Tenant.scala

```diff
@@ -201,44 +237,125 @@ object Tenant {
   }
 
   private def editInfo(
-      state: Tenant.EstablishedTenantState,
+      state: Tenant.EstablishedTenant,
       editInfoCommand: EditInfo,
-  ): Either[Error, TenantEnvelope] = {
-    val infoToUpdate = editInfoCommand.infoToUpdate
-    val stateInfo = state.info
-    val newInfo = infoFromEditableInfo(infoToUpdate, stateInfo)
+  ): Either[Error, TenantEnvelope] = state match {
+    case draftState: DraftTenant =>
+      val infoToUpdate = editInfoCommand.infoToUpdate
+      val stateInfo = draftState.info
```

Comment:
For clarity it really helps to have a consistent naming grammar. The inconsistency here is that "state" is used as an adjective prefix in one case and as a noun (also at the beginning of the name) in the second case. Avoiding using the same word with different meaning helps to make names clearer.

This article might be interesting: https://betterprogramming.pub/a-useful-framework-for-naming-your-classes-functions-and-variables-e7d186e3189f I don't necessarily follow that to the letter but it has a lot of good ideas about how to name identifiers in a communicative way.

Reply:
Decided to change `stateOpt` to `establishedInfoOpt` in order to avoid renaming all `state` objects for consistency. If new name is too long can be just `establishedOpt` or `infoOpt`